In [56]:
import pandas as pd
from astropy import units as u
from astropy.coordinates import SkyCoord
import numpy as np

In [57]:
data = pd.read_csv('Outliers.csv')

In [58]:
runcat, ra, dec = data['runcat'],data['ra'],data['dec']

In [59]:
results = []

In [60]:
for i, val in enumerate(ra):
    coordinates = SkyCoord(ra= val*u.degree, dec=dec[i]*u.degree)
    results.append([runcat[i],coordinates.to_string('hmsdms')])
    print(coordinates)

<SkyCoord (ICRS): (ra, dec) in deg
    (275.09136281, 7.18563978)>
<SkyCoord (ICRS): (ra, dec) in deg
    (275.22669226, 7.28275854)>
<SkyCoord (ICRS): (ra, dec) in deg
    (274.70727893, 6.47906455)>
<SkyCoord (ICRS): (ra, dec) in deg
    (275.30495118, 7.41051643)>
<SkyCoord (ICRS): (ra, dec) in deg
    (274.54375643, 7.2904081)>
<SkyCoord (ICRS): (ra, dec) in deg
    (275.12532555, 6.5720707)>
<SkyCoord (ICRS): (ra, dec) in deg
    (274.46923647, 6.77732602)>
<SkyCoord (ICRS): (ra, dec) in deg
    (274.87458706, 7.78413362)>
<SkyCoord (ICRS): (ra, dec) in deg
    (274.98029633, 7.73851434)>
<SkyCoord (ICRS): (ra, dec) in deg
    (274.51119529, 7.21252494)>


In [61]:
for i, val in enumerate(ra):
    coordinates = SkyCoord(ra= val*u.degree, dec=dec[i]*u.degree)
    print(runcat[i],coordinates.ra.degree, coordinates.dec.degree)
    results.append([runcat[i],coordinates.to_string('hmsdms')])

108 275.09136281196277 7.18563977734883
78 275.2266922562669 7.282758536153431
164 274.7072789327543 6.479064548812852
66 275.30495117845055 7.410516432223797
485 274.5437564287246 7.290408098919997
97 275.12532554850543 6.572070695523553
197 274.46923647342635 6.777326024469214
136 274.87458706368153 7.7841336180177
121 274.98029633034804 7.738514343509355
191 274.5111952850805 7.212524936724012


In [78]:
results = np.array(results)
final_results = []
#print(results) 
for val in results: 
    line = val[1]
    #print(line)
    line = line.split()
    temp=[]
    temp.append(val[0])
    for coord in line: 
        #print(coord)
        coord = coord.replace('h',':')
        coord = coord.replace('m',':')
        coord = coord.replace('s','')
        coord = coord.replace('d',':')
        coord = coord.replace('m',':')
        coord = coord.replace('s','')
        temp.append(coord)
    final_results.append(temp)

print(final_results)

[['108', '18:20:21.92707487', '+07:11:08.30319846'], ['78', '18:20:54.4061415', '+07:16:57.93073015'], ['164', '18:18:49.74694386', '+06:28:44.63237573'], ['66', '18:21:13.18828283', '+07:24:37.85915601'], ['485', '18:18:10.50154289', '+07:17:25.46915611'], ['97', '18:20:30.07813164', '+06:34:19.45450388'], ['197', '18:17:52.61675362', '+06:46:38.37368809'], ['136', '18:19:29.90089528', '+07:47:02.88102486'], ['121', '18:19:55.27111928', '+07:44:18.65163663'], ['191', '18:18:02.68686842', '+07:12:45.08977221'], ['108', '18:20:21.92707487', '+07:11:08.30319846'], ['78', '18:20:54.4061415', '+07:16:57.93073015'], ['164', '18:18:49.74694386', '+06:28:44.63237573'], ['66', '18:21:13.18828283', '+07:24:37.85915601'], ['485', '18:18:10.50154289', '+07:17:25.46915611'], ['97', '18:20:30.07813164', '+06:34:19.45450388'], ['197', '18:17:52.61675362', '+06:46:38.37368809'], ['136', '18:19:29.90089528', '+07:47:02.88102486'], ['121', '18:19:55.27111928', '+07:44:18.65163663'], ['191', '18:18:02.6

108: MAXI 
78: OLD 186 
164: NVSSS J18... 
66: ONE 0F THE LOBES OF THE BLACK HOLES 
485: NOTHING THERE
97: NVSS J18...
197: NVSS ...
136: NOTHING THERe
121: VERY FAINT BUT VARIABLE SOURCE
191: FAINT BUT VARIABLE









In [37]:
coordinates = SkyCoord(ra= ra*u.degree, dec=dec*u.degree)

UnitTypeError: Angle instances require units equivalent to 'rad', but no unit was given.

In [16]:
print(coordinates)
coordinates.to_string('hmsdms')

<SkyCoord (ICRS): (ra, dec) in deg
    (275.12532555, 6.5720707)>


'18h20m30.07813164s +06d34m19.45450388s'

In [17]:
def truncate_float(float_number, decimal_places):
    multiplier = 10 ** decimal_places
    return int(float_number * multiplier) / multiplier

In [82]:
for i, val in enumerate(ra):
    coordinates = SkyCoord(ra= val*u.degree, dec=dec[i]*u.degree)
    print(runcat[i],round(coordinates.ra.degree,3))
"""     if runcat[i] == 108:
        print(truncate_float(coordinates.ra.degree,4))
        print(truncate_float(coordinates.dec.degree,4))
        print(coordinates) """

275.091
275.0913
7.1856
<SkyCoord (ICRS): (ra, dec) in deg
    (275.09136281, 7.18563978)>
275.227
274.707
275.862
275.305
274.544
275.125
275.305
274.011
274.469
274.875
275.9
274.98
276.158
274.511
275.166
275.11
275.077
275.31
275.189
274.169
275.217
275.424
275.12
276.009
275.24
274.477
275.478
275.347
274.893
275.113
274.897
275.01
275.306
275.221
273.996
274.773
275.011
274.376
275.751
274.306
274.96
275.28
275.326
274.307
275.098
274.131
275.43
274.419
274.548
275.307
275.168
275.93
274.524
275.398
275.149
275.872
274.748
275.385
275.816
274.912
275.774
275.73
275.604
275.169
274.351
275.429
275.258
275.615
274.327
275.014
274.987
274.634
275.447
275.485
274.338
274.633
275.67
274.594
274.769
275.164
275.511
274.985
274.766
275.348
274.067
274.731
274.86
275.19
274.846
274.715
275.397
275.404
274.839
275.957
275.439
275.092
274.58
274.932
274.827
275.649
275.496
274.94
274.846
274.609
275.345
274.796
275.098
274.474
274.836
275.674
275.458
275.029
276.172
275.043
274.495
274.727

In [79]:
ra_155, dec_155 = 275.166, 7.145

data = pd.read_csv('Wholedatasetoutput.csv')
runcat, ra, dec = data['runcat'],data['ra'],data['dec']

In [80]:
for i, val in enumerate(ra):
    coordinates = SkyCoord(ra= val*u.degree, dec=dec[i]*u.degree)
    #print(round(coordinates.ra.degree,3))
    #print(truncate_float(coordinates.ra.degree,3))
    #print(truncate_float(coordinates.dec.degree,3))
    if truncate_float(coordinates.ra.degree,3) == ra_155 and truncate_float(coordinates.dec.degree,3) == dec_155:
        print(runcat[i])

In [81]:
for i, val in enumerate(runcat):
    if val == 136:
        print()